src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -2701,6 +2725,22 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
      * However T must be fully defined before we type the instantiation, as it'll end up as a parent type,
      * which must be fully defined. Would be nice to have some kind of mechanism to insert type vars in a block of code,
      * and have the instantiation of the first occurrence propagate to the rest of the block.
+     *
+     * TODO: by-name params
```

Comment:
Not sure what you're referring to, but I meant that `f(println("!"))` should behave the same as `f.accept(println("!"))` below.